include/swift/AST/DiagnosticsSema.def

```diff
@@ -2662,6 +2662,11 @@ ERROR(inheritance_from_cf_class,none,
 ERROR(inheritance_from_objc_runtime_visible_class,none,
       "cannot inherit from class %0 because it is only visible via the "
       "Objective-C runtime", (Identifier))
+ERROR(composition_of_protocol_and_anyobject,none,
```

Comment:
I see there is this tailored diagnostic for composition but not a regression test case for exercise it. We should add the respective test :)